### DISCUSSION TITLE
Post-Release Ideas

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3322] [Development]
After a release is a good time to make big changes, since there is less development going on.

- [Rename Context](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438163)
- [Repo Main Branch](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438167)
- [Increase the Build Minimum Versions](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438169)
- [Clang Format New Options](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438178)
- [Cull Old Issues](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438181)
- [Cull Old Members](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438183)
- [Create Author Repo](https://github.com/neomutt/neomutt/discussions/3322#discussioncomment-2438185)

Do you have any big ideas?

**Comment by flatcap** [comment]
# Rename Context

It took years to do, but the global `Context` is dead.

It's time to rename `struct Context` to `struct MailboxView`.
The new name clearly describes what the struct does.

**Comment by flatcap** [comment]
# Repo Main Branch

Rename the `master` branch to `main` for all repos.

I've already updated all the "lesser" repos.
Altering the big ones will mean fixing pull-requests and lots of workflow scripts.

- https://github.com/neomutt/corpus-address
- https://github.com/neomutt/neomutt
- https://github.com/neomutt/neomutt.github.io

**Reply by aliceinwire** [reply]
what's the profit of using main?

**Reply by flatcap** [reply]
Consistency, nothing else.
`git`, since 2.31, defaults to `main` as do GitHub and GitLab.

In the end, switching over turned out to be pretty simple.

**Reply by aliceinwire** [reply]
ah I didn't know. thanks for the information :+1:

**Comment by flatcap** [comment]
# Increase the Build Minimum Versions

- C99 -> C11
- POSIX:2001 -> POSIX:2008

These standards are over a decade old.

As jindraj recently discovered, NeoMutt doesn't even compile against decade-old versions of ncurses.

**Comment by flatcap** [comment]
# Clang Format New Options

Add some new clang-format options to make the code a little neater.
They prevent lines from ending in `=` or `(` at the expense of making some lines slightly longer.

- PenaltyBreakAssignment: 100
- PenaltyBreakBeforeFirstCallParameter: 100
- BreakStringLiterals: false

Across the entire codebase, they save about 600 lines.

See the changes, here: 41fb9ce8

**Comment by flatcap** [comment]
# Cull Old Issues

There are lots of issues we've abandoned, often due to being unreproducible.

Time to close them, unless the original poster can reproduce the bug in the latest version.

**Comment by flatcap** [comment]
# Cull Old Members

This is usually a unmentioned part of my project management.

NeoMutt is a very welcoming project: If you contribute, you can become a member.

However, if you haven't contributed anything for six months and you haven't even said, "Hello!" in IRC, then you'll get evicted.

The last release was five months ago.

**Comment by flatcap** [comment]
# Create Author Repo

Currently, the central place for all the Authors/Contributors is the git dir in the [management repo](https://github.com/neomutt/management).

I'd like to split this out into a new "authors" repo.
One central place to list everyone that's contributed to the NeoMutt project.

I'd also like to remove the `AUTHORS.md` and `.mailmap` from the `neomutt` repo.
We'd list a few principal authors/maintainers and link to the full list.

This would reduce the maintenance effort.
